SimDiagramBuilder, WorldSimTreeBuilder, and RobotPlanInterpolator relocated

naveenoid opened the pull request:
Components relocated to drake/manipulation

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/6864)
<!-- Reviewable:end -->

naveenoid commented:
+@sammy-tri for feature review.

---

Review status: 0 of 22 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrlHl-Tuytu2UYnqONA:bp61js4)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
(tagging @tristanthrush)

---

Review status: 0 of 22 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrlHpxsFngZrAoIC43S:b-z8mz62)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
I think your CI problem is due to not updating `CMakeLists.txt` which referenced the files you're using?  Not 100% sure.

Apologies for digging into some code which wasn't actually changed in this PR, but we'd been setting the bar kinda low for code in examples/ for a while, and I don't necessarily want to carry all of that into `manipulation/`

---

Reviewed 30 of 30 files at r1.
Review status: all files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 181 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZkxTuo_fRgJg2q4C:-KrlZkxUlyS2-wW-etIc:bb2ig14) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L181)):*
> ```cpp
>   place_location.translation() = post_locations[0] + post_height_offset;
>   place_location.linear() =
>       Matrix3<double>(AngleAxis<double>(M_PI / 2., Vector3<double>::UnitZ()));
> ```

BTW there's a very good chance that these unnecessary whitespace changes will cause merge problems for me (not saying you have to undo it in this PR, but it's another reason to consider avoiding it)

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 11 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlVdkLyhjruxSm2LLm:-KrlVdkMpvu8TkHKm7vy:bcvqst0) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L11)):*
> ```cpp
> namespace manipulation {
> namespace planner {
> namespace test {
> ```

BTW This is inconsistent with the other test in this directory which does not use an additional "test" namespace (it's also inconsistent with the tests this PR moves into `manipulation/util/test`.  From looking around (briefly) there are other tests which use `namespace test`, but it seems mostly to be used by test utilities

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlTFGXkm34E5PkbpZZ:-KrlTFGYv-mHj7OFN9Gm:b-rnhpht) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L83)):*
> ```cpp
> 
>   robotlocomotion::robot_plan_t plan =
>       examples::kuka_iiwa_arm::EncodeKeyFrames(dut.tree(), t, info, q);
> ```

I don't really like this dependency on example code.  Can we either move EncodeKeyFrames or build the robot_plan_t directly here?  The latter should be fairly simple with only one joint.  (sorry, I know I did this in the first place)

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 112 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlU4eOee6pPA53bhI1:-KrlU4ePDBy-YPmuT5um:bqh06tu) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L112)):*
> ```cpp
>     const double velocity =
>         output->get_vector_data(dut.get_state_output_port().get_index())
>             ->GetAtIndex(kNumJoints);
> ```

BTW intermingled whitespace changes which are not required by the style guide or fixing a style issue enforced in CI make diffs harder to read (IMO).  (no need to undo it in this PR or anything, but...)

---

*[drake/manipulation/util/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZHfRVnGfDKHVtE4p:-KrlZHfRVnGfDKHVtE4q:b-l7zluu) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L15)):*
> ```Python
> 
> drake_cc_library(
>     name = "world_sim_tree_builder",
> ```

This doesn't seem to have it's own test.  It should either get one or we should explain why it has adequate test coverage.

---

*[drake/manipulation/util/BUILD, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYXiA0sajysxMjzBP:-KrlYXiA0sajysxMjzBQ:bqornc7) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L24)):*
> ```Python
>         "//drake/multibody:rigid_body_tree_construction",
>         "//drake/multibody/parsers",
>         "//drake/multibody/rigid_body_plant:drake_visualizer",
> ```

I think several of these are unused (see `world_sim_tree_builder.cc` comments)

---

*[drake/manipulation/util/sim_diagram_builder.h, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYqBwX0QBfIfE_5ou:-KrlYqBxSjUN9MVdnnTI:b-2fup3y) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/sim_diagram_builder.h#L6)):*
> ```objc
> #include <unordered_map>
> #include <utility>
> #include <vector>
> ```

unused (I think?)

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYEpA7ACwZqyFu8u0:-KrlYEpA7ACwZqyFu8u1:b-2fup3y) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L10)):*
> ```cpp
> 
> #include "drake/common/find_resource.h"
> #include "drake/math/roll_pitch_yaw.h"
> ```

unused (I think?)

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXaOBNv5R3eFBK-e2:-KrlXaOC_oy_Awgt8hy3:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L15)):*
> ```cpp
> #include "drake/multibody/parsers/urdf_parser.h"
> #include "drake/multibody/rigid_body_frame.h"
> #include "drake/multibody/rigid_body_plant/drake_visualizer.h"
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXEYTo9Y-stWTowDs:-KrlXEYUEiZl3mxNBYDO:b7pdx0b) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L21)):*
> ```cpp
> using Eigen::aligned_allocator;
> using Eigen::Vector3d;
> using drake::multibody::joints::FloatingBaseType;
> ```

we use this type, but in a way that this `using` has no effect

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXGvDCb0Ycd6WNwAb:-KrlXGvDCb0Ycd6WNwAc:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L22)):*
> ```cpp
> using Eigen::Vector3d;
> using drake::multibody::joints::FloatingBaseType;
> using drake::multibody::joints::kFixed;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXMNt9Q5cvyTm94tp:-KrlXMNt9Q5cvyTm94tq:b7pdx0b) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L24)):*
> ```cpp
> using drake::multibody::joints::kFixed;
> using drake::multibody::joints::kQuaternion;
> using drake::parsers::ModelInstanceIdTable;
> ```

we use this type, but in a way that this `using` has no effect

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXU3d0aBpvqdERsZy:-KrlXU3eOplHe9frAaDP:b7pdx0b) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L25)):*
> ```cpp
> using drake::multibody::joints::kQuaternion;
> using drake::parsers::ModelInstanceIdTable;
> using drake::parsers::sdf::AddModelInstancesFromSdfFile;
> ```

we use this type, but in a way that this `using` has no effect

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXUqel1cJuzAhDY5_:-KrlXUqel1cJuzAhDY5a:b7pdx0b) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L26)):*
> ```cpp
> using drake::parsers::ModelInstanceIdTable;
> using drake::parsers::sdf::AddModelInstancesFromSdfFile;
> using drake::parsers::urdf::AddModelInstanceFromUrdfFile;
> ```

we use this type, but in a way that this `using` has no effect

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXq37nPWaKvYH-6Vb:-KrlXq39TvKmRnwupxNf:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L27)):*
> ```cpp
> using drake::parsers::sdf::AddModelInstancesFromSdfFile;
> using drake::parsers::urdf::AddModelInstanceFromUrdfFile;
> using drake::systems::Context;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXrCOnBVz8xmNdryk:-KrlXrCPWw--tzDvxY-a:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L28)):*
> ```cpp
> using drake::parsers::urdf::AddModelInstanceFromUrdfFile;
> using drake::systems::Context;
> using drake::systems::ContinuousState;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 29 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXscODuIufyyLAyCQ:-KrlXscODuIufyyLAyCR:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L29)):*
> ```cpp
> using drake::systems::Context;
> using drake::systems::ContinuousState;
> using drake::systems::DrakeVisualizer;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 30 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXv-brXc1RvELGKnw:-KrlXv-brXc1RvELGKnx:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L30)):*
> ```cpp
> using drake::systems::ContinuousState;
> using drake::systems::DrakeVisualizer;
> using drake::systems::VectorBase;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYKh-xKMB3j4ui3sh:-KrlYKh0p3jI9LKb4JjL:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L32)):*
> ```cpp
> using drake::systems::VectorBase;
> using std::allocate_shared;
> using std::make_unique;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYLgWQmNStLpe_a2Z:-KrlYLgWQmNStLpe_a2_:b-jmepyd) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L34)):*
> ```cpp
> using std::make_unique;
> using std::string;
> using std::unique_ptr;
> ```

unused

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 77 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlWcNhR8kM9QaXRh0i:-KrlWcNiyoJvmdAkELD4:bdq34wb) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L77)):*
> ```cpp
>     const drake::multibody::joints::FloatingBaseType floating_base_type) {
>   DRAKE_DEMAND(!built_);
>   unused(xyz, rpy);  // TODO(jwnimmer-tri) This seems bad.
> ```

I think we need to fix this TODO before moving this code out of examples, either by implementing the missing functionality or dropping the misleading arguments entirely.

---

*[drake/manipulation/util/test/sim_diagram_builder_test.cc, line 88 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlVMXjhwUPnDdXQbLn:-KrlVMXkbvXUpN5blPNT:bmb2osy) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/test/sim_diagram_builder_test.cc#L88)):*
> ```cpp
>   // Adds a controller.
>   VectorX<double> iiwa_kp, iiwa_kd, iiwa_ki;
>   examples::kuka_iiwa_arm::SetPositionControlledIiwaGains(&iiwa_kp, &iiwa_ki,
> ```

Do the actual gains make any difference to the outcome of this test?  If not, can we choose something arbitrary here instead of creating a dependency on example code?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrlUhj76Mz9pyqkNrfO:b6c7lh3)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Strongly agree with your comment on upgrading coding quality with this move. About the CMakeLists, can we just get rid of it? I realistically dont see anyone using it anymore. Lastly, whitespace changes were introduced by clang-format  - isnt that the defacto recommendation? Moving on to fix the defects now.

---

Review status: all files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-Krlirntrx4x_jLhtKaw:b-e49csk)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
My perspective is that `clang-format` is one way to produce code which will pass the CI style tests.  (as mentioned in http://drake.mit.edu/code_style_tools.html ).  I've been trying to avoid clang formatting whole files as part of other changes (as opposed to just formatting the lines you've changed like in https://dx13.co.uk/articles/2015/4/3/Setting-up-git-clang-format.html ).  There's no writeup in drake's docs yet on how to set up something like `git-clang-format`...  my personal thought is that if the tests pass there's no reason to run `clang-format` and avoiding mixing functional and formatting changes isn't too painful.

Not sure about the `CMakeLists.txt` question.  As long as there are no downstream users, deleting the moved files should be fine.

---

Review status: all files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrllQkuE8HifiZb5773:boiuqar)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
Review status: 12 of 24 files reviewed at latest revision, 22 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 181 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZkxTuo_fRgJg2q4C:-KrmCR2Pi4cv7trSOsUe:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L181)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW there's a very good chance that these unnecessary whitespace changes will cause merge problems for me (not saying you have to undo it in this PR, but it's another reason to consider avoiding it)
</blockquote></details>

Done.

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 11 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlVdkLyhjruxSm2LLm:-KrmDqg1OIQSqTKDJAgY:b-wnfmk9) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L11)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW This is inconsistent with the other test in this directory which does not use an additional "test" namespace (it's also inconsistent with the tests this PR moves into `manipulation/util/test`.  From looking around (briefly) there are other tests which use `namespace test`, but it seems mostly to be used by test utilities
</blockquote></details>

Done. Removed.

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlTFGXkm34E5PkbpZZ:-KrmUaokCCWHpdAGowC4:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L83)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I don't really like this dependency on example code.  Can we either move EncodeKeyFrames or build the robot_plan_t directly here?  The latter should be fairly simple with only one joint.  (sorry, I know I did this in the first place)
</blockquote></details>

Done.

---

*[drake/manipulation/planner/test/robot_plan_interpolator_test.cc, line 112 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlU4eOee6pPA53bhI1:-Krq43F-IgIN84BZvzdk:bb6sh2i) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/planner/test/robot_plan_interpolator_test.cc#L112)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW intermingled whitespace changes which are not required by the style guide or fixing a style issue enforced in CI make diffs harder to read (IMO).  (no need to undo it in this PR or anything, but...)
</blockquote></details>

Ack.

---

*[drake/manipulation/util/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZHfRVnGfDKHVtE4p:-Krq4xYwnobbNm5uJgyo:b-b62iv4) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L15)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

This doesn't seem to have it's own test.  It should either get one or we should explain why it has adequate test coverage.
</blockquote></details>

It currently being jointly tested with the SimDiagramBuilder. Either I leave a todo here or quickly whip up a test within this PR. Your call.

---

*[drake/manipulation/util/BUILD, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYXiA0sajysxMjzBP:-Krq3u2eXAt2dPDAJP6L:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L24)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I think several of these are unused (see `world_sim_tree_builder.cc` comments)
</blockquote></details>

Done.

---

*[drake/manipulation/util/sim_diagram_builder.h, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYqBwX0QBfIfE_5ou:-Krpu-AiO5SyOxuNlxqr:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/sim_diagram_builder.h#L6)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused (I think?)
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYEpA7ACwZqyFu8u0:-KrptzGRsEa1uLN770Rd:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L10)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused (I think?)
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXaOBNv5R3eFBK-e2:-KrptzXU3r_ok1ONVLiq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L15)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXEYTo9Y-stWTowDs:-Krq-qp8cO6eUVxt7x26:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L21)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

we use this type, but in a way that this `using` has no effect
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXGvDCb0Ycd6WNwAb:-KrpuFPKP55XWA99KWiz:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L22)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXMNt9Q5cvyTm94tp:-KrpuFE0WgNWh7v6JZof:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L24)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

we use this type, but in a way that this `using` has no effect
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXU3d0aBpvqdERsZy:-KrpuFiSV0N4M95pgAGa:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L25)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

we use this type, but in a way that this `using` has no effect
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXUqel1cJuzAhDY5_:-KrpuG38SKKfGm0Xlplj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L26)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

we use this type, but in a way that this `using` has no effect
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXq37nPWaKvYH-6Vb:-KrpuGyofi8huo6-CtWZ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L27)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 28 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXrCOnBVz8xmNdryk:-KrpuHVJ9ohL0m6VRrSu:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L28)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 29 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXscODuIufyyLAyCQ:-KrpuHkR5TjXTozp8z_2:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L29)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 30 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXv-brXc1RvELGKnw:-KrpuKQLKhho4qGht_1k:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L30)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 32 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYKh-xKMB3j4ui3sh:-KrpuN2_cy18cRGQLXzV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L32)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlYLgWQmNStLpe_a2Z:-KrpuOhkd_NM_lFCHYak:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L34)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 77 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlWcNhR8kM9QaXRh0i:-KrpxMpjgPD6kISb-XOY:b-wnfmk9) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L77)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I think we need to fix this TODO before moving this code out of examples, either by implementing the missing functionality or dropping the misleading arguments entirely.
</blockquote></details>

Done. Removed.

---

*[drake/manipulation/util/test/sim_diagram_builder_test.cc, line 88 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlVMXjhwUPnDdXQbLn:-Krq-WkH6AL1P0aMu9mh:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/test/sim_diagram_builder_test.cc#L88)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Do the actual gains make any difference to the outcome of this test?  If not, can we choose something arbitrary here instead of creating a dependency on example code?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> pending my last couple of nits.  +@sherm1 for platform review.

---

Reviewed 5 of 30 files at r1, 12 of 12 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/CMakeLists.txt, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqFCjN10lEhBjCTQwt:-KrqFCjOcfwvLyy2aJWJ:blge19) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/CMakeLists.txt#L26)):*
> ```CMake
> 
> 
> #if(lcm_FOUND)
> ```

Adding this large commented out block seems strange.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 43 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqGkH-pryueqvsGtqS:-KrqGkH0e1i5s7cX_em6:buysxkj) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L43)):*
> ```cpp
> DEFINE_bool(quick, false,
>             "Run only a brief simulation and return success "
>                 "without executing the entire task");
> ```

BTW This indentation looks really weird to me (if somehow that's what clang-format produced, I guess it's OK since it's the tool of record...)

---

*[drake/manipulation/util/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZHfRVnGfDKHVtE4p:-KrqFtkZiVyXDeDz3xgU:bhzvd0l) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L15)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

It currently being jointly tested with the SimDiagramBuilder. Either I leave a todo here or quickly whip up a test within this PR. Your call.
</blockquote></details>

I think I'm OK with the combined test coverage as long as we note that somewhere.

---

*[drake/manipulation/util/BUILD, line 20 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqFnAaFBYiJyqaGZiG:-KrqFnAbzF_QxuwquTjN:bheg2du) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/manipulation/util/BUILD#L20)):*
> ```Python
>     deps = [
>         "//drake/common:find_resource",
>         "//drake/common:unused",
> ```

I believe this is now unused.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXEYTo9Y-stWTowDs:-KrqG76DYXWBo6Ok0JcM:bocxtk7) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L21)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't see the change?

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXGvDCb0Ycd6WNwAb:-KrqG9vvg4FNauzBuoWs:bocxtk7) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L22)):*
<details><summary><i>Previously, naveenoid (Naveen Kuppuswamy) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't see the change?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrqHAlLO1p6HHEmPMQe:b-26rj5k)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Platform <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, with a few not-very-weighty BTWs, ptal. Most are about inconsistent placement of `using`s in `.cc` files. Either the outermost or innermost namespace would be the usual placement; there is kind of a random assortment here.

---

Reviewed 20 of 30 files at r1, 12 of 12 files at r2.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/CMakeLists.txt, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqFCjN10lEhBjCTQwt:-KrqP0HAp0fTqzrTxYVi:b-798myw) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/CMakeLists.txt#L26)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Adding this large commented out block seems strange.
</blockquote></details>

BTW please remove this block unless there is a strong reason it has to be here.

---

*[drake/examples/kuka_iiwa_arm/iiwa_controller.cc, line 32 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPKUCcjU45271c1eM:-KrqPKUCcjU45271c1eN:b-mmx456) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_controller.cc#L32)):*
> ```cpp
> 
> namespace drake {
> using manipulation::planner::RobotPlanInterpolator;
> ```

BTW can this `using` be placed a few lines lower, in the anonymous namespace? Or, it can go at the top of the file. Either seems tidier/more conventional.

---

*[drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc, line 46 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPWeSuBPeES6XOygg:-KrqPWeSuBPeES6XOygh:b-qyzujt) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc#L46)):*
> ```cpp
> namespace drake {
> using manipulation::util::WorldSimTreeBuilder;
> using manipulation::util::ModelInstanceInfo;
> ```

BTW please move these to be with the other `using manipulation::` stuff 6 lines below (in the anonymous namespace).

---

*[drake/examples/kuka_iiwa_arm/kuka_simulation.cc, line 38 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPm-6LpM2bRJIa9Wd:-KrqPm-6LpM2bRJIa9We:b-m4wwki) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/kuka_simulation.cc#L38)):*
> ```cpp
> 
> namespace drake {
> using manipulation::util::SimDiagramBuilder;
> ```

BTW please move to be with the other `using`s below.

---

*[drake/examples/kuka_iiwa_arm/controlled_kuka/controlled_kuka_demo.cc, line 42 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPvsElxoN99hdIVmv:-KrqPvsElxoN99hdIVmw:b-gx5p9u) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/controlled_kuka/controlled_kuka_demo.cc#L42)):*
> ```cpp
> 
> namespace drake {
> using manipulation::util::SimDiagramBuilder;
> ```

BTW move down?

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 43 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqGkH-pryueqvsGtqS:-KrqQOCc855oD_UCOYG8:b-qnf7mx) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L43)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW This indentation looks really weird to me (if somehow that's what clang-format produced, I guess it's OK since it's the tool of record...)
</blockquote></details>

BTW agreed. Nicer if the left quotes line up. No defect though.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQDYJElygGo75wfex:-KrqQDYJElygGo75wfey:bhy5tse) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L55)):*
> ```cpp
> using manipulation::util::ModelInstanceInfo;
> using manipulation::planner::RobotPlanInterpolator;
> using manipulation::util::WorldSimTreeBuilder;
> ```

BTW all these `using`s would be better a few lines below in the anonymous namespace.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.cc, line 32 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQfnFVXVlZGXs_Xk9:-KrqQfnFVXVlZGXs_XkA:b-o1hsar) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.cc#L32)):*
> ```cpp
> using systems::OutputPort;
> using systems::RigidBodyPlant;
> using manipulation::util::SimDiagramBuilder;
> ```

BTW can these be moved into the innermost namespace?

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.h, line 15 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQlSCBD3nXnTL8wNO:-KrqQlSCBD3nXnTL8wNP:b-6dz1np) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.h#L15)):*
> ```objc
> namespace drake {
> using systems::PassThrough;
> using manipulation::util::ModelInstanceInfo;
> ```

BTW move usings down?

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/multi_arm_with_gripper_demo.cc, line 15 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQr_UDRgAMz-rtRgZ:-KrqQr_UDRgAMz-rtRg_:b-wb25d1) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/multi_arm_with_gripper_demo.cc#L15)):*
> ```cpp
> using manipulation::util::ModelInstanceInfo;
> using manipulation::util::WorldSimTreeBuilder;
> using manipulation::util::SimDiagramBuilder;
> ```

BTW move usings down into anonymous namespace?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrqS7IK-yTDPw7QAN8O:bra99ux)*
<!-- Sent from Reviewable.io -->

naveenoid commented:
If all seems well, please go ahead and squash-merge. I wont have time to look at this PR again today.

---

Review status: 12 of 24 files reviewed at latest revision, 14 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/CMakeLists.txt, line 26 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqFCjN10lEhBjCTQwt:-KrqiFNN5kabEMuxqEmH:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/CMakeLists.txt#L26)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW please remove this block unless there is a strong reason it has to be here.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_controller.cc, line 32 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPKUCcjU45271c1eM:-KrqiLGJvNIbH7nn8Xma:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_controller.cc#L32)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW can this `using` be placed a few lines lower, in the anonymous namespace? Or, it can go at the top of the file. Either seems tidier/more conventional.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc, line 46 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPWeSuBPeES6XOygg:-KrqiTdq7cdddRXuYLL1:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_wsg_simulation.cc#L46)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW please move these to be with the other `using manipulation::` stuff 6 lines below (in the anonymous namespace).
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/kuka_simulation.cc, line 38 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPm-6LpM2bRJIa9Wd:-KrqiZVVKUu09w0PJND6:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/kuka_simulation.cc#L38)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW please move to be with the other `using`s below.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/controlled_kuka/controlled_kuka_demo.cc, line 42 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqPvsElxoN99hdIVmv:-KrqidqB33vNmBYOm1ao:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/controlled_kuka/controlled_kuka_demo.cc#L42)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW move down?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 43 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqGkH-pryueqvsGtqS:-Krqj1d1W4exFznMlNUI:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L43)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW agreed. Nicer if the left quotes line up. No defect though.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQDYJElygGo75wfex:-Krqjn5w9Dqe48-Fpl40:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L55)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW all these `using`s would be better a few lines below in the anonymous namespace.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.cc, line 32 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQfnFVXVlZGXs_Xk9:-Krqjsb1cZDPSM8p8MeD:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.cc#L32)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW can these be moved into the innermost namespace?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.h, line 15 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQlSCBD3nXnTL8wNO:-KrqjwRH_tv6oYTdeeOK:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/iiwa_wsg_diagram_factory.h#L15)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW move usings down?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/iiwa_world/multi_arm_with_gripper_demo.cc, line 15 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqQr_UDRgAMz-rtRgZ:-Krqk07AFy9tM4rI2W6J:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/examples/kuka_iiwa_arm/iiwa_world/multi_arm_with_gripper_demo.cc#L15)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW move usings down into anonymous namespace?
</blockquote></details>

Done.

---

*[drake/manipulation/util/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlZHfRVnGfDKHVtE4p:-Krqh3ST0zRFUvbLdtM1:b-yzvge6) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/BUILD#L15)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I think I'm OK with the combined test coverage as long as we note that somewhere.
</blockquote></details>

Done. Added on top of the SimDiagramBuilderTest.

---

*[drake/manipulation/util/BUILD, line 20 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrqFnAaFBYiJyqaGZiG:-KrqhmcnOCnMDbJnXQPb:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/84b3df1fc9c011e7c35ba2360ddb0f064a3eaf4d/drake/manipulation/util/BUILD#L20)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I believe this is now unused.
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXEYTo9Y-stWTowDs:-KrqhyqODBLQSr8AD4Ce:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L21)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I don't see the change?
</blockquote></details>

Done.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-KrlXGvDCb0Ycd6WNwAb:-Krqi730VjkZye_PzijA:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8636c7ab245d5fa000bcdb76721bda43bd402b13/drake/manipulation/util/world_sim_tree_builder.cc#L22)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I don't see the change?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-KrqkXCjrgS9sxFBzzeJ:bnk8ecg)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 2 of 30 files at r1, 12 of 12 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Everything looks good, but CI failed apparently due to linting problems.

---

Reviewed 2 of 30 files at r1, 12 of 12 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6864#-:-Krr1llt_ByW1O-pV1FN:b-2295ir)*
<!-- Sent from Reviewable.io -->
